File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSTypeClassDeclarationImpl.java
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import com.intellij.psi.PsiElementVisitor;
import net.kenro.ji.jin.purescript.psi.PSPositionedDeclarationRef;
import net.kenro.ji.jin.purescript.psi.PSTypeClassDeclaration;
import net.kenro.ji.jin.purescript.psi.PSVisitor;
import org.jetbrains.annotations.NotNull;

public class PSTypeClassDeclarationImpl extends PSPsiElement implements PSTypeClassDeclaration {

    public PSTypeClassDeclarationImpl(final ASTNode node) {
        super(node);
    }

    public void accept(@NotNull final PSVisitor visitor) {
        visitor.visitPSTypeClassDeclaration(this);
    }

    public void accept(@NotNull final PsiElementVisitor visitor) {
        if (visitor instanceof PSVisitor) accept((PSVisitor) visitor);
        else super.accept(visitor);
    }





}

File: src/main/java/net/kenro/ji/jin/purescript/psi/impl/PSTypeClassDeclarationImpl.kt
package net.kenro.ji.jin.purescript.psi.impl;

import com.intellij.lang.ASTNode;
import net.kenro.ji.jin.purescript.psi.PSTypeClassDeclaration;

public class PSTypeClassDeclarationImpl extends PSPsiElement implements PSTypeClassDeclaration {

    public PSTypeClassDeclarationImpl(final ASTNode node) {
        super(node);
    }

}
